pkg/provisioners/message_dispatcher.go

```diff
@@ -223,3 +224,41 @@ func (d *MessageDispatcher) resolveURL(destination string, defaultNamespace stri
 		Path:   "/",
 	}
 }
+
+// ComputeRoutes sends the message to the router.
+func (d *MessageDispatcher) ComputeRoutes(message *Message, router string, defaults DispatchDefaults) ([]string, error) {
+	routerURL := d.resolveURL(router, defaults.Namespace)
+	req, err := http.NewRequest(http.MethodPost, routerURL.String(), bytes.NewReader(message.Payload))
```

Comment:
A CoudEvent router will only need the attributes of the CloudEvent context. Sending the complete message to the router will not be necessary in all cases.

Reply:
It's necessary for content-based routing. Are you suggesting the router should advertise what attributes it needs (could initially vs coarse-grained, context vs content vs both)?

Reply:
I realize that there is also content-based routing, but the Broker concept so far completely relies on CloudEvents. After all, this is exactly what CloudEvents are designed for. To make it short, yes. 😄 
I know that in this stage we should not optimize too much. I only brought it up, because routing based on the context attributes is a major use case.